src/fragments/start/getting-started/reactnative/auth.mdx

```diff
@@ -0,0 +1,383 @@
+```
+
+</Block>
+<Block name="React Native CLI">
+
+```bash
+yarn add @aws-amplify/ui-react-native react-native-safe-area-context react-native-get-random-values react-native-url-polyfill
+```
+
+You will also need to install the pod dependencies for iOS:
+
+```sh
+npx pod-install
+```
+
+</Block>
+</BlockSwitcher>
+
+### Add the Amplify UI Authenticator component
+
+<BlockSwitcher>
+<Block name="Expo CLI">
+
+Open **App.js** and make the following changes:
+
+</Block>
+<Block name="React Native CLI">
+
+Open **App.tsx** and make the following changes:
+
+</Block>
+</BlockSwitcher>
+
+1. Import the `withAuthenticator` Higher-Order Component and `useAuthenticator` hook:
+
+```javascript
+
+import { withAuthenticator, useAuthenticator } from '@aws-amplify/ui-react-native';
+
+````
+
+2. The `useAuthenticator` hook can be added to any component that needs it. Create a custom `SignOutButton` component that will be used within the App component, and destructure the `{ user, signOut }` properties returned in the context.
+
+```javascript
+
+const SignOutButton = () => {
+  const { user, signOut } = useAuthenticator((context) => [context.user]);
+  return (
+    <Pressable onPress={signOut} style={styles.buttonContainer}>
+      <Text style={styles.buttonText}>Hello, {user.username}! Click here to sign out!</Text>
+    </Pressable>
+  )
+};
+
+```
+
+3. Add the `SignOutButton` component to  your `App` component:
+
+```javascript
+
+// ...
+return (
+  <View style={styles.container}>
+    <SignOutButton />
+//...
+
+```
+
+4. Lastly, wrap your `App` export with the `withAuthenticator` Amplify UI component:
+
+```javascript
+
+export default withAuthenticator(App);
+
+```
+
+<BlockSwitcher>
+<Block name="Expo CLI">
+
+Run the app to see the new authentication flow protecting the app:
+
+```bash
+npm start
+```
+
+Now you should see the app load with an authentication flow allowing users to sign up and sign in. Below you can find a full sample of the code:
+
+```javascript
+import React, {useEffect, useState} from 'react';
+import {
+  StyleSheet,
+  Text,
+  View,
+  TextInput,
+  Pressable,
+  SafeAreaView,
+} from 'react-native';
+import {API, graphqlOperation} from 'aws-amplify';
+import {createTodo} from './src/graphql/mutations';
+import {listTodos} from './src/graphql/queries';
+import {
+  withAuthenticator,
+  useAuthenticator,
+} from '@aws-amplify/ui-react-native';
+
+import { Amplify } from 'aws-amplify';
+import awsExports from './src/aws-exports';
+Amplify.configure(awsExports);
+
+const SignOutButton = () => {
+  const {user, signOut} = useAuthenticator(context => [context.user]);
+  return (
+    <Pressable onPress={signOut} style={styles.buttonContainer}>
+      <Text style={styles.buttonText}>
+        Hello, {user.username}! Click here to sign out!
+      </Text>
+    </Pressable>
+  );
+};
+
+const initialState = {name: '', description: ''};
```

Comment:
Since the `Authenticator` has an `initialState` prop, suggest naming this something else to prevent confusion

Reply:
changed to `initialFormState`